### ISSUE TITLE
Hoogle it for unknown symbol

### ISSUE THREAD

**Issue by July541** [haskell/haskell-language-server#3187]
As the image shows, I hope we can add a code action to hoogle it for unknown symbols

<img width="798" alt="image" src="https://user-images.githubusercontent.com/16057632/190862300-7eb32dc5-1ecb-4075-a1a9-eb55c9631129.png">

**Comment by sloorush** [comment]
Very useful, but will it get chaotic sometimes and may feel spammy to the user?

**Comment by July541** [comment]
> Very useful, but will it get chaotic sometimes and may feel spammy to the user?

People can disable it anytime, and I think there won't be so many unknown symbols at once.

**Comment by sloorush** [comment]
> > Very useful, but will it get chaotic sometimes and may feel spammy to the user?
> 
> People can disable it anytime, and I think there won't be so many unknown symbols at once.

Yep! Then this will be pretty useful and fast for the users who want it.

**Comment by pepeiborra** [comment]
Why only unknown symbols? I might want to Google any identifier that doesn't have documentation or a link to Hackage. Which brings me to the second point, I suspect a hover is a better fit than a code action for this.

**Comment by July541** [comment]
@jneira told me we have some attempts for hoogle integration, it's fine to have a hover. But for unknown symbols, I just want to redirect users to the hoogle web page because of worrying about the size of hoogle.

https://twitter.com/jneira/status/1571232027583602690

**Comment by pepeiborra** [comment]
> @jneira told me we have some attempts for hoogle integration, it's fine to have a hover. But for unknown symbols, I just want to redirect users to the hoogle web page because of worrying about the size of hoogle.
> 
> https://twitter.com/jneira/status/1571232027583602690

I don't think adding a Hoogle code action that will pop up for every single squiggly is going to be well received by users. On the other hand, a hover that presents a Hoogle link for identifiers that lack documentation *might* be useful

**Comment by July541** [comment]
> On the other hand, a hover that presents a Hoogle link for identifiers that lack documentation might be useful.

Strongly agree with this.

> I don't think adding a Hoogle code action that will pop up for every single squiggly is going to be well received by users.

Following your thought, what should be displayed while hovering on unknown symbols like `getFist` in the image?

**Comment by pepeiborra** [comment]
> Following your thought, what should be displayed while hovering on unknown symbols like getFist in the image?

Same as for known symbols without docs, hover should show a link to Hoogle it.

I should also point out that the screenshot that you shared in the issue description is very different from what I see in my HLS installation. Do you not have the ghcide plugins enabled? Maybe you are on 9.4?

<img width="558" alt="image" src="https://user-images.githubusercontent.com/26626/190892329-86be20d4-b0dd-45a9-a2ca-d07d705ab12a.png">

**Comment by July541** [comment]
Yes, I'm on 9.4.

Emm, sounds great! So you are thinking to add a link like https://hoogle.haskell.org/?hoogle=getFirst for any symbols without docs(including unknown symbols)?
